### PR TITLE
55 Add ARRL Sweepstakes Contest

### PR DESCRIPTION
**Pull request by w7sst**
Implements a simulation of the ARRL Sweepstakes Contest. Entering the exchange is similar to N1MM Logger+ where the exchange components can be entered in either order, either `<serial><precedence> <check><section>` or `<check><section> <serial><precedence>`.

The entered callsign can be corrected in the exchange field. If the sender corrects their callsign, it can be corrected in the exchange field without changing fields.

Also, the `<check><section>` of the calling station will be occasionally placed into the exchange field. This is similar to N1MM behavior where the `<check><section>` is taken from the call history file. The cursor is then placed after the `<check><section>` allowing the user to enter the `<serial><precedence>` without moving the cursor. If the sender sends a different `<section>`, it can be corrected at the end of the exchange field.

### CODE REVIEW

File: Readme.txt [hunk 1]
```diff
@@ -469,6 +469,48 @@ CONTEST INFORMATION
     F: Emergency Operation Centers
     Rules: https://contests.arrl.org/ContestRules/Field-Day-Rules.pdf
 
+    ARRL Sweepstates
```
Comment (w7sst) [inline Readme.txt:472]: I fixed the spelling error: `Sweepstates` to `Sweepstakes`

File: Readme.txt [hunk 2]
```diff
@@ -469,6 +469,48 @@ CONTEST INFORMATION
...
+    - Callsigns corrections can be optionally entered into the Exchange field.
+      If the caller sends a corrected callsign, you can enter the updated call
+      anywhere in the exchange field without changing focus to the original
+      callsign entry field.  For example, entering "W1AW 123 A 38 CT" will
+      update the callsign when storing the QSO into the log.
+    - For added realism, the simulation will populate the exchange field
+      with the Check and Section entries for the entered callsign. This
+      behavior matches current N1MM behavior. However, the simulation will
+      occasionally modify the Section to simulate that the operator is
+      operating from a different location.
```
Comment (w7sst) [inline Readme.txt:512]: I updated this last paragraph to:
```
    - For added realism, the simulation will populate the exchange field
      with the Check and Section entries for the entered callsign. This
      behavior matches current N1MM behavior. In addition, the simulation will
      occasionally modify the Section to simulate when an operator is operating
      from a different location.

      By default, the Check/Section value will be pre-populated in the Exchange
      field about 50% of the time. This behavior can be adjusted between 0% and
      100% by setting the following keyword in the MorseRunner.ini file:
        [Settings]
        ShowCheckSection=50
      Valid values range between 0 and 100, representing 0% and 100%.
      Note that a zero value will disable this feature.
```

**Comment by K6OK** [comment]
I ran 1.85-dev.4 in both Pileup and Single Call mode.  Works great!  Only have a couple of comments:

The label for the rightmost entry field says Nr Prec Call CK Sect.  Suggest delete "Call" because it implies the call is a mandatory part of the exchange to be entered, which is not the case.  I understand call correction can happen here but that can be explained in a help file or wiki.

For the log, maybe programmatically reduce the font size so there aren't so many "..." column headers?   73 Jim K6OK

**Comment by w7sst** [comment]
> For the log, maybe programmatically reduce the font size so there aren't so many "..." column headers? 73 Jim K6OK

Jim, would you please include a screen shot showing the "..." in your column headers? (You can paste an image into this box and github will store it for you. I use Window's Snip to grab screenshots and then paste them here.)

On my laptop, I am not seeing any "..." in my headers. Perhaps I'm programming to my screen and not all screens in general. Have you had issues with other contests as well? Has this behavior (the "..."'s) recently started? Do you see it with other contests? If you do see it with other contests, can you try a prior version to see if you see the '...'s. If so, that is a good data point.

Thank you,
Mike W7SST

@K6OK

**Comment by K6OK** [comment]
> Jim, would you please include a screen shot showing the "..." in your column headers?

Mike --  This is from my desktop.  1920x1080 at the default 100% scale.   
image

**Comment by w7sst** [comment]
I think I found it. The font size is 9 for the log area. The font size for all other controls in the UI is 7. I'll change the ListView to 7 and we can see if that helps. I'll get a new build out tonight sometime.

**Comment by w7sst** [comment]
> I think I found it. The font size is 9 for the log area. The font size for all other controls in the UI is 7. I'll change the ListView2.Font.Size to 7 and we can see if that helps. I'll get a new build out tonight sometime.

However, making the Log report smaller font is not good long term. I'll spend some time over the next few days to see if I can programmatically increase the width of the report columns. If that doesn't work, I can at least reduce the font a little.

**Comment by w7sst** [comment]
@K6OK 
Hi Jim, I created Issue #344 to track this new issue with the column headers.

**Pull request merged**